include/deal.II/lac/parallel_vector.templates.h

```diff
@@ -308,6 +308,11 @@ namespace parallel
     Vector<Number>::compress_start (const unsigned int counter,
                                     ::dealii::VectorOperation::values operation)
     {
+      // These parameters are not used for some MPI and debug settings.
+      int ierr;
```

Comment:
What's the reason for pulling the declaration of ierr  out to the function level here and elsewhere?